Can't release maven-java with JDK 17

Updated to JDK 17 in https://github.com/lightbend/kalix-jvm-sdk/pull/1431

```
[INFO] Kalix Java Maven parent pom ........................ FAILURE [  2.296 s]
...
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------: Unable to make field private final java.util.Comparator java.util.TreeMap.comparator accessible: module java.base does not "opens java.util" to unnamed module @25d3cfc8
[ERROR]
```

It's in maven-java: `mvn -P release -B deploy`

I tried building locally but then some other problem with the gpg plugin.

Maybe we are just using old versions, I haven't looked or investigated further.

--- a/maven-java/pom.xml
+++ b/maven-java/pom.xml
@@ -58,7 +58,7 @@
         <plugin>
           <groupId>org.sonatype.plugins</groupId>
           <artifactId>nexus-staging-maven-plugin</artifactId>
-          <version>1.6.7</version>
+          <version>1.6.13</version>
           <extensions>true</extensions>
           <configuration>
             <serverId>ossrh</serverId>

